### PR TITLE
Bump prefect to 0.15.1 and upgrade other dependencies

### PR DESCRIPTION
**Pull request by mikix**
I went through the [changelog](https://docs.prefect.io/api/latest/changelog.html#_0-15-0) and it does not appear that we use any of the now-removed (or now-deprecated) features in edx-prefectutils or prefect-flows.

### CODE REVIEW

File: requirements/test.in [hunk 1]
```diff
@@ -8,7 +8,7 @@ Sphinx==1.8.5
 twine==1.14.0
 pytest==4.6.5
 pytest-runner==5.1
-httpretty
+httpretty<1.1.0  # can remove constraint once https://github.com/gabrielfalcao/HTTPretty/issues/425 is fixed
```
Comment (mikix) [inline requirements/test.in:11]: Calling out this change - this and the `base.in` change to bump prefect to 0.15.1 are the only manual changes made. Everything else was just `make upgrade`

**Review by brianhw — APPROVED** [review]
I also went through the log and didn't see anything problematic. 👍

**Comment by jazibhumayun** [comment]
I am using `prefect auth login -t` command which uses tokens. I think that will be deprecated with v0.15.
I will need to create a new auth key and do some Key Vault changing.

**Comment by jazibhumayun** [comment]
[https://github.com/edx/edx-internal/pull/5273/files](https://github.com/edx/edx-internal/pull/5273/files)
and [https://github.com/edx-ops/analytics-secure/pull/587](https://github.com/edx-ops/analytics-secure/pull/587) should also be merged along with prefect v0.15 version upgrade PRs

**Pull request merged**